### PR TITLE
added arangodb authentication using username and password

### PR DESCRIPTION
**Pull request by ajayguls**
Added support for arangodb authentication using username and password

### CODE REVIEW

File: adapter.go [hunk 1]
```diff
@@ -110,6 +120,11 @@ func NewAdapter(options ...adapterOption) (persist.Adapter, error) {
 	if err != nil {
 		return nil, err
 	}
+	if a.dbUser != "" {
+		fmt.Println("using db username password as %s, %s", a.dbUser, a.dbPasswd)
```
Comment (adamwasila) [inline adapter.go]: I guess we can live without this line especially that it is logging sensitive data.

File: adapter.go [hunk 2]
```diff
@@ -65,6 +67,14 @@ func OpDatabaseName(dbName string) func(*adapter) {
 	}
 }
 
+// OpDatabaseCreds configures username and password of database used; default is ""
+func OpDatabaseCreds(user, passwd string) func(*adapter) {
```
Comment (adamwasila) [inline adapter.go]: As different types of authentication are possible it would be nice to explicitly state type in name, eg. `OpBasicAuthCredentials()`.

**Review by adamwasila — CHANGES_REQUESTED** [review]
Many thanks. Please fix these minor issues I found and it's ready to merge.

File: adapter.go (hunk 1 above)
Comment (ajayguls) [reply adapter.go]: removed.

File: adapter.go (hunk 2 above)
Comment (ajayguls) [reply adapter.go]: fixed

**Comment by ajayguls** [comment]
Have handled both comments and updated the PR.

**Comment by adamwasila** [comment]
Ok, LGTM

**Pull request merged**